prereise/cli/data_sources/solar_data.py

```diff
@@ -0,0 +1,182 @@
+        data = ga_wind.retrieve_data(
+            solar_plants, key, start_date=start_date, end_date=end_date
+        )
+        data.to_pickle(file_path)
+
+
+class SolarDataNationalSolarRadiationDatabase(DataSource):
+    @property
+    def command_name(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.command_name`
+
+        :return: (*str*)
+        """
+        return "solar_data_nsrdb"
+
+    @property
+    def command_help(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.command_help`
+
+        :return: (*str*)
+        """
+        return "Download solar data from the National Solar Radiation Database"
+
+    @property
+    def extract_arguments(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.extract_arguments`
+
+        :return: (*str*)
+        """
+        return [
+            {
+                "command_flags": ["--region", "-r"],
+                "required": True,
+                "choices": REGION_CHOICES,
+                "type": str,
+                "action": "append",
+            },
+            {
+                "command_flags": ["--method", "-m"],
+                "required": True,
+                "choices": [NAIVE_STRING, SAM_STRING],
+                "type": str,
+            },
+            {
+                "command_flags": ["--year", "-y"],
+                "required": True,
+                "type": validate_year,
+                "help": "The start date for the data download in format 'YYYY'",
+            },
+            {
+                "command_flags": ["--file_path", "-fp"],
+                "required": True,
+                "type": validate_file_path,
+                "help": FILE_PATH_HELP_STRING,
+            },
+            {
+                "command_flags": ["--email", "-e"],
+                "required": True,
+                "type": str,
+                "help": "Email used to sign up at https://developer.nrel.gov/signup/",
+            },
+            {
+                "command_flags": ["--key", "-k"],
+                "required": True,
+                "type": str,
+                "help": API_KEY_HELP_STRING,
+            },
+        ]
+
+    def extract(self, region, method, year, file_path, email, key, **kwargs):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.extract`
+
+        :param list region: list of regions to download data for
+        :param str method: string indicating the modeling for power output
+        :param str year: string in the format YYYY denoting year to download from
+        :param str file_path: file location on local filesystem on where to store the data
+        :param str email: email used to sign up at https://developer.nrel.gov/signup/
+        :param str key: API key that can requested at https://developer.nrel.gov/signup/
+        """
+        regions = list(set(region))
```

Comment:
@rouille maybe we should handle removing duplicates and/or support passing a set [here](https://github.com/Breakthrough-Energy/PowerSimData/blob/develop/powersimdata/network/usa_tamu/model.py#L77)?

Reply:
yes we should

Reply:
PR for that: https://github.com/Breakthrough-Energy/PowerSimData/pull/510

Reply:
@goccert25 fyi that is merged, so we can eliminate any `list(set(region))` since it will be handled within the `Grid` constructor

Reply:
awesome. I was trying to think of solutions for the constant `list(set(region))` copy paste but this solves it nicely